types/fabric/fabric-impl.d.ts

```diff
@@ -142,15 +142,79 @@ interface IDataURLOptions {
 	withoutShadow?: boolean;
 }
 
+/**
+ * A general Event type that represents all events mixed in a single type.
+ */
 interface IEvent {
-	e: Event;
+  /**
+   * The native Event instance, for example, a DOM MouseEvent if the user selected or 
+   * modified fabric.js objects using the mouse in the browser. 
+   */
+  e: Event;
 	target?: Object;
-    subTargets?: Object[],
+  subTargets?: Object[];
 	button?: number;
+  selected?: Object[];
 	isClick?: boolean;
+  deselected: Object[];
```

Comment:
Since these properties aren't on every single event, they should only be present in the subclasses you wrote.

If you have an object that's typed as an `IEvent` and need a property from a subclass you would cast it instead.
```ts
let event: IEvent = ...
(event as ISelectionClearedEvent).deselected
```